### DISCUSSION TITLE
Using Prince in PowerShell - cannot parse C:\Program Files (x86)

### DISCUSSION THREAD

**Discussion by sophielamb** [thegetty/quire#135] [Get Help] [answered]
I am not able to generate the static pdf files for our project. I installed Prince the same time as installing Quire on my Windows machine.

When I run `quire pdf` Prince kicks into gear and sets about generating the file, but it cannot read the program file location. On installation it went into the default `C:\\Program Files (x86)`. The spaces mean it can't be read. 

```
PS C:\Users\sophi> Set-Location -Path C:\\Users\sophi\Documents\Materia\Issue-1-Spring-21-q-01
PS C:\Users\sophi\Documents\Materia\Issue-1-Spring-21-q-01> quire pdf
\ Building PDFBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
- Building PDF'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
× Error: Command failed: C:\Program Files (x86)\Prince\engine\bin\prince.exe http://localhost:1313// http://localhost:1313//contents http://localhost:1313//intro http://localhost:1313//articles/article-jd-rh-et http://localhost:1313//articles/article-gp-mla-vm http://localhost:1313//articles/article-jll-sac-fc http://localhost:1313//articles/article-my http://localhost:1313//articles/article-ej http://localhost:1313//about/about http://localhost:1313//about/acknowledgements -o static\downloads\output.pdf --no-warn-css-unsupported --no-warn-css-unknown --no-warn-css
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.

PS C:\Users\sophi\Documents\Materia\Issue-1-Spring-21-q-01>
```

Is there a way to get PowerShell to recognise the location of the Prince folder? Have other win users had similar problems, or use Prince differently?

The closest workaround I could think of is to enclose the path name in quotation marks, but that's not possible in this situation.

best wishes,
Sophie

**Comment by geealbers** [comment]
HI @sophielamb, what I'm seeing here is that Prince is looking for the publication pages at the wrong URLs. For instance, it lists `http://localhost:1313//intro` when it should be `http://localhost:1313/intro` (with only a single slash). This might be a Windows pathing issue in the Quire code (which is something we've run into before) rather than an issue with your Prince installation. @hbalenda, what do you think?

Sophie, I don't have my PC with me to test it on my end, but I can when I'm back at work on Monday. Could you also try testing your Prince installation by pointing it at another URL directly? For instance, if you run this command, do you get a `balloons.pdf` file created in your home directory?

```
prince https://en.wikipedia.org/wiki/Balloon -o balloons.pdf
```

In the meantime, Hannah might have some insight for us. She's one of Getty's developers working on Quire.

**Reply by sophielamb** [reply]
Thanks for looking into this. I copied the command and received an error containing the same double slash in the path:

```
prince : The term 'prince' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
At line:1 char:1
+ prince https://en.wikipedia.org/wiki/Balloon -o balloons.pdf
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (prince:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

**Comment by geealbers** [comment, marked as answer]
Just to close the loop on this issue, it's been fixed with the release of Quire v0.19.2. Anyone interested can read more in the [pull request notes](https://github.com/thegetty/quire/pull/137), but essentially we had to make a small update in the [Quire CLI](https://quire.getty.edu/documentation/quire-commands/) to the commands for PDF generation to get them to work for users of the PowerShell shell application on Windows.
